dynawo/sources/Models/CPP/ModelNetwork/DYNModelDanglingLine.cpp

```diff
@@ -678,15 +679,31 @@ ModelDanglingLine::defineElements(std::vector<Element> &elements, std::map<std::
 
 void
 ModelDanglingLine::evalZ(const double& t) {
-  z_[0] = static_cast<double> (connectionState_);
-  z_[1] = currentLimitsDesactivate_;
+  if (doubleNotEquals(z_[1], getCurrentLimitsDesactivate())) {
+    setCurrentLimitsDesactivate(z_[1]);
+    Trace::debug() << DYNLog(DeactivateCurrentLimits, id_) << Trace::endline;
+  }
 
   if (currentLimits_) {
     ModelCurrentLimits::state_t currentLimitState;
     currentLimitState = currentLimits_->evalZ(id(), t, &(g_[0]), network_, currentLimitsDesactivate_);
     if (currentLimitState == ModelCurrentLimits::COMPONENT_OPEN)
       z_[0] = OPEN;
   }
+
+  State currState = static_cast<State>(z_[0]);
+  if (currState != connectionState_) {
+    Trace::debug() << DYNLog(DanglingLineStateChange, id_, connectionState_, currState) << Trace::endline;
+    stateModified_ = true;
+    if (connectionState_ == CLOSED) {
+      network_->addEvent(id_, DYNTimeline(DanglingLineConnected));
+      modelBus_->getVoltageLevel()->connectNode(modelBus_->getBusIndex());
+    } else if (connectionState_ == OPEN) {
+      network_->addEvent(id_, DYNTimeline(DanglingLineDisconnected));
+      modelBus_->getVoltageLevel()->disconnectNode(modelBus_->getBusIndex());
+    }
+    connectionState_ = static_cast<State>(z_[0]);
+  }
```

Comment:
I find it strange to have z_[0] handled after z_[1], even if it doesn't change the behavior. I don't see any apparent reason to change the order here.